### PR TITLE
Adds docket and appeal methods for distribution

### PR DESCRIPTION
**Pull request by amprokop**
Resolves https://github.com/department-of-veterans-affairs/caseflow/issues/9139

### CODE REVIEW

File: app/models/docket.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    if priority == true
+      scope.merge(Appeal.ordered_by_distribution_ready_date)
+    else
+      scope.order("receipt_date")
+    end
+  end
+
+  def count(priority: nil, ready: nil)
+    appeals(priority: priority, ready: ready).length
```
Comment (cmgiven) [inline app/models/docket.rb]: Does `length` need to instantiate all of the appeals? Or does it smartly count in the database?

Comment (amprokop) [reply app/models/docket.rb]: Good point, it instantiates all the appeals — but `count` does not behave here like i expected, it returns a hash and not a number. I'll look into it.

Comment (amprokop) [reply app/models/docket.rb]: ah, this is why, I believe — https://api.rubyonrails.org/classes/ActiveRecord/Calculations.html#method-i-count

Comment (amprokop) [reply app/models/docket.rb]: Sigh. I think `appeals(priority: priority, ready: ready).count.length` would actually return what we want here. `appeals(priority: priority, ready: ready).count` returns
```
{[65, 65]=>2,
 [66, 66]=>2,
 [67, 67]=>2,
 [68, 68]=>2,
 [69, 69]=>2,
 [70, 70]=>2,
 [71, 71]=>2,
 [72, 72]=>2,
 [73, 73]=>2,
 [74, 74]=>2,
 [75, 75]=>2,
 [76, 76]=>2,
 [77, 77]=>2,
 [78, 78]=>2,
 [79, 79]=>2,
 [80, 80]=>2,
 [81, 81]=>2,
 [82, 82]=>2,
 [83, 83]=>2,
 [84, 84]=>2}
```

File: app/models/appeal.rb [hunk 2]
```diff
@@ -44,7 +44,21 @@ class Appeal < DecisionReview
   scope :ready_for_distribution, lambda {
     joins(:tasks)
       .group("appeals.id")
-      .having("count(case when tasks.type = ? and tasks.status = ? then 1 end) = ?", "DistributionTask", "assigned", 1)
+      .having("count(case when tasks.type = ? and tasks.status = ? then 1 end) = ?",
```
[the remark this answers is not in the document]
Comment (amprokop) [reply app/models/appeal.rb]: Yeah, I was looking into that. You're right, I think it's harmless and safer to add the `>=1`here.

File: app/models/docket.rb [hunk 3]
```diff
@@ -0,0 +1,89 @@
+class Docket
+  include ActiveModel::Model
+
+  def docket_type
+    fail Caseflow::Error::MustImplementInSubclass
+  end
+
+  def appeals(priority: nil, ready: nil)
+    fail "'ready for distribution' value cannot be false" if ready == false
```
Comment (aroltsch) [inline app/models/docket.rb]: Why are you passing this as a parameter if it can only be `true`

File: app/models/docket.rb [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+    scope = docket_appeals
+    scope = scope.merge(Appeal.ready_for_distribution) if ready == true
+    scope = scope.merge(Appeal.all_priority) if priority == true
```
Comment (aroltsch) [inline app/models/docket.rb]: You could perhaps combine lines 11-20 into:

```
if priority == true
   scope = scope.merge(Appeal.all_priority)
   scope.merge(Appeal.ordered_by_distribution_ready_date)
   return scope
end

scope = scope.merge(Appeal.all_nonpriority)
scope.order("receipt_date")
```

File: app/models/docket.rb (hunk 3 above)
Comment (cmgiven) [reply app/models/docket.rb]: `ready` can also be `nil`. That's the default.

Comment (amprokop) [reply app/models/docket.rb]: it can be `true` or `nil`

File: app/models/docket.rb (hunk 4 above)
Comment (amprokop) [reply app/models/docket.rb]: yeah that sounds nice, i can do that

**Pull request merged**